content/en/blog/2021/better-external-authz/index.md

```diff
@@ -0,0 +1,545 @@
+data:
+  mesh: |-
+    # Add the following contents:
+    extensionProviders:
+    - name: "opa.local"
+      envoyExtAuthzGrpc:
+        service: "local-opa-grpc.local"
+        port: "9191"
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< tab name="Deploy OPA in a separate pod" category-value="opa-standalone" >}}
+
+{{< text yaml >}}
+apiVersion: v1
+data:
+  mesh: |-
+    # Add the following contents:
+    extensionProviders:
+    - name: "opa.default"
+      envoyExtAuthzGrpc:
+        service: "opa.default.svc.cluster.local"
+        port: "9191"
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< /tabset >}}
+
+### Create AuthorizationPolicy with CUSTOM action
+
+Run the following command to create the authorization policy that enables the external authorization on all paths
+except `/ip`:
+
+{{< tabset category-name="opa-deploy" >}}
+
+{{< tab name="Deploy OPA in the same pod" category-value="opa-same" >}}
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: httpbin-opa
+spec:
+  selector:
+    matchLabels:
+      app: httpbin-with-opa
+  action: CUSTOM
+  provider:
+    name: "opa.local"
+  rules:
+  - to:
+    - operation:
+        notPaths: ["/ip"]
+EOF
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< tab name="Deploy OPA in a separate pod" category-value="opa-standalone" >}}
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: httpbin-opa
+spec:
+  selector:
+    matchLabels:
+      app: httpbin
+  action: CUSTOM
+  provider:
+    name: "opa.default"
+  rules:
+  - to:
+    - operation:
+        notPaths: ["/ip"]
+EOF
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< /tabset >}}
+
+### Test the OPA policy
+
+1. Create the sleep pod to send the request:
+
+    {{< text bash >}}
+    $ kubectl apply -f @samples/sleep/sleep.yaml@
+    $ export SLEEP_POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})
+    {{< /text >}}
+
+1. Use a test JWT token signed by the OPA:
+
+    {{< text bash >}}
+    $ export TOKEN_PATH_HEADERS="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiTDJobFlXUmxjbk09IiwibmJmIjoxNTAwMDAwMDAwLCJleHAiOjE5MDAwMDAwMDB9.9yl8LcZdq-5UpNLm0Hn0nnoBHXXAnK4e8RSl9vn6l98"
+    {{< /text >}}
+
+    The test JWT token has the following claims:
+
+    {{< text json >}}
+    {
+      "path": "L2hlYWRlcnM=",
+      "nbf": 1500000000,
+      "exp": 1900000000
+    }
+    {{< /text >}}
+
+    The `path` claim has value `L2hlYWRlcnM=` which is the base64 encode of `/headers`.
+
+1. Send request to path `/headers` without token, should be rejected with 403 because there is no JWT token:
+
+    {{< text bash >}}
+    $ kubectl exec ${SLEEP_POD} -c sleep  -- curl http://httpbin-with-opa:8000/headers -s -o /dev/null -w "%{http_code}\n"
+    403
+    {{< /text >}}
+
+1. Send request to path `/get` with valid token, should be rejected with 403 because the path `/get` is not matched with the token `/headers`:
```

Comment:
```suggestion
1. Send a request to path `/get` with a valid token. This should be rejected with 403 because the path `/get` is not matched with the token `/headers`:
```